File: cli/src/main/java/com/gmail/marcosav2010/module/ModuleCommandRegistry.java
package com.gmail.marcosav2010.module;

import com.gmail.marcosav2010.command.CommandRegistry;

import java.lang.annotation.*;

@Documented
@Retention(value = RetentionPolicy.RUNTIME)
@Target(value = ElementType.TYPE)
public @interface ModuleCommandRegistry {

    Class<? extends CommandRegistry> value();
}

File: cli/src/main/java/com/gmail/marcosav2010/module/ModuleCommandRegistry.kt
package com.gmail.marcosav2010.module

import com.gmail.marcosav2010.command.CommandRegistry
import kotlin.reflect.KClass

@MustBeDocumented
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class ModuleCommandRegistry(val value: KClass<out CommandRegistry>)